src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -407,16 +414,42 @@ else if (getRun() != null) {
                 this.failedSince = getRun().getNumber();
             } else {
                 LOGGER.warning("trouble calculating getFailedSince. We've got prev, but no owner.");
-                // failedSince will be 0, which isn't correct. 
+                // failedSince will be 0, which isn't correct.
             }
         }
         return failedSince;
     }
-    
+
+    /**
+     * If this test was skipped, then return the build number
+     * when this test was first skipped.
+     */
+    @Exported(visibility=9)
+    public int getSkippedSince() {
+        // If we haven't calculated failedSince yet, and we should,
+        // do it now.
+        if (skippedSince==0 && getSkipCount()==1) {
+            CaseResult prev = getPreviousResult();
+            if(prev!=null && !prev.isPassed())
+                this.skippedSince = prev.getSkippedSince();
```

Comment:
What happens if in `previous` the test failed and therefore it has been disabled now? Then `previous.isPassed `is `false`, too. Might be a good setup for an integration test.

Reply:
You're absolutely right - when a test becomes skipped, its skippedSince counter is set to the current build number.
In a subsequent build:
- If the test passes or fails, the skippedSince counter will be reset to 0.
- If the test is skipped again, the skippedSince is carried over from the previous build.